Maybe.Test/Maybe.Test.csproj

```diff
@@ -1,21 +1,21 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```

Comment:
Ideally we want to have the upgrade to .NET 8 as a separate PR. Also we need to check what (if any) compatibility  there is with this upgrade. Does it require a major release? Probably.

Reply:
This is just in the test project, doesn't affect the actual library

Reply:
We probably want to open another PR to do this upgrade. Can we postpone these changes when we do that? It's confusing to have a project where only some of the csproj have been upgraded.

Reply:
The lib actually targets netstandard, which is already multiple versions and different form the test project target, I don't think we need to upgrade it.

I can reverse these changes, but a new PR would upgrade just the test project and wouldn't even generate a release

Reply:
Let's reverse all lib upgrades in this PR for now. That includes the package references in this csproj (like FluentAssertions) as well as the `azure-pipelines.yaml` changes. Let's focus only on the changes related to the name of this PR.